server/middlewares/frontendMiddleware.js

```diff
@@ -0,0 +1,55 @@
+const express = require('express');
+const path = require('path');
+
+// Dev middleware
+const addDevMiddlewares = (app, options) => {
+  const webpackDevMiddleware = require('webpack-dev-middleware');
+  const webpackHotMiddleware = require('webpack-hot-middleware');
+  const webpack = require('webpack');
+
+  const compiler = webpack(options);
+  const middleware = webpackDevMiddleware(compiler, {
+    noInfo: true,
+    publicPath: options.output.publicPath,
+    lazy: true,
+    silent: true,
+  });
+
+  app.use(middleware);
+  app.use(webpackHotMiddleware(compiler));
+
+  // Since webpackDevMiddleware uses memory-fs internally to store build
+  // artifacts, we use it instead
+  const fs = middleware.fileSystem;
+
+  app.get('*', (req, res) => {
+    const file = fs.readFileSync(path.join(compiler.outputPath, 'index.html'));
+    res.send(file.toString());
+  });
+};
+
+const addProdMiddlewares = (app, options) => {
```

Comment:
Well, in the scope of this middleware, I don't think so. We could add stuff like 500 page for uncaught exceptions, but it's more a fit for `server.js`.